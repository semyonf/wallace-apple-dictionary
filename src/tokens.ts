export const tokens = {
  logger: 'logger',
  taskQueue: 'task-queue',
  wallaceWikiParser: 'wallace-wiki-parser',
  pageDOMLoader: 'page-dom-loader',
  scraper: 'scraper',
} as const;
