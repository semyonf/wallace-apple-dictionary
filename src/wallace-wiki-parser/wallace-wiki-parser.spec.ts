import { WallaceWikiParser } from './wallace-wiki-parser';
import { JSDOM } from 'jsdom';
import { loadResource, ResourceName } from '../test-helpers/resource-loader';
import { testInjector } from '../test-helpers/test-injector';
import { tokens } from '../tokens';

const wallaceWikiParser = testInjector.resolve(tokens.wallaceWikiParser);

describe(WallaceWikiParser.name, () => {
  it('should be able to parse dictionary entries from a page', async () => {
    const pageHTML = await loadResource(
      ResourceName.RawPageWithDictionaryEntries,
    );
    const pageDOM = new JSDOM(pageHTML, { url: 'http://localhost' });

    const parsedDictionaryEntries =
      wallaceWikiParser.parseDictionaryEntries(pageDOM);

    expect(parsedDictionaryEntries).toMatchSnapshot();
  });

  it('should be able to parse table of contents', async () => {
    const pageHTML = await loadResource(ResourceName.RawHomePage);
    const pageDOM = new JSDOM(pageHTML, { url: 'http://localhost' });

    const parsedTableOfContents =
      wallaceWikiParser.parseTableOfContents(pageDOM);

    expect(parsedTableOfContents).toMatchSnapshot();
  });
});
