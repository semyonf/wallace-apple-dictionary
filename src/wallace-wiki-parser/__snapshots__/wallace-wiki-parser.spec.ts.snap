// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WallaceWikiParser should be able to parse dictionary entries from a page 1`] = `
Array [
  Object {
    "definition": "Hal is referring to the fact that the office he's in is decorated with art by Frederic Remington (1861-1909), an American painter whose work can be seen online here.",
    "pageName": "Page 3",
    "term": "Remington-hung",
  },
  Object {
    "definition": "Hal's Uncle, Charles Tavis, is head of the Enfield Tennis Academy.",
    "pageName": "Page 3",
    "term": "Uncle Charles",
  },
  Object {
    "definition": "A type of knot used to tie a necktie.  Picture of a half-Windsor here.",
    "pageName": "Page 3",
    "term": "Half-Windsors",
  },
  Object {
    "definition": "See prorector. Possibly an originally German term. Also see prorector index entry.",
    "pageName": "Page 3",
    "term": "prorector",
  },
  Object {
    "definition": "See deLint index entry; p. 4.",
    "pageName": "Page 3",
    "term": "A. deLint",
  },
  Object {
    "definition": "Uncle Charles, mentioned previously. See Page 50.",
    "pageName": "Page 3",
    "term": "C.T.",
  },
  Object {
    "definition": "fringe; outer boundary",
    "pageName": "Page 3",
    "term": "periphery",
  },
  Object {
    "definition": "Hal's full first name is given for the first time.",
    "pageName": "Page 3",
    "term": "Harold Incandenza",
  },
  Object {
    "definition": "A fictional town just west of Boston, where parts of the real Boston neighborhoods of Brighton and Allston exist in reality. There used to be a real Enfield in western Massachusetts but it was disincorporated in 1938 and flooded by the creation of the Quabbin Reservoir.",
    "pageName": "Page 3",
    "term": "Enfield",
  },
  Object {
    "definition": "like a tennis court, presumably",
    "pageName": "Page 3",
    "term": "court-shaped",
  },
  Object {
    "definition": "Organization of North American Nations Collegiate Athletic Association -- presumably the future complement of the NCAA.
        ",
    "pageName": "Page 4",
    "term": "O.N.A.N.C.A.A.",
  },
  Object {
    "definition": "\\"A benign encysted tumor of the skin, esp. on the scalp, containing sebaceous matter; a sebaceous cyst\\" (Random House Unabridged Dictionary).",
    "pageName": "Page 4",
    "term": "wen",
  },
  Object {
    "definition": "Enfield Tennis Academy prorector. See deLint index entry.",
    "pageName": "Page 4",
    "term": "Aubrey F. deLint",
  },
  Object {
    "definition": "Asserts as true or alleges.",
    "pageName": "Page 4",
    "term": "avers",
  },
  Object {
    "definition": "as in, among the cream of the crop, or among the very best.",
    "pageName": "Page 4",
    "term": "among the very cream",
  },
  Object {
    "definition": "The Randolph Tennis Center is a real place, near Tucson, Ariz. and the main campus of the University of Arizona.",
    "pageName": "Page 4",
    "term": "Randolph Tennis Center",
  },
  Object {
    "definition": "\\"El Con\\" is short for \\"El Conquistador,\\" and while there is a Hilton El Conquistador Hotel in Tucson, the Marriot has a different name.",
    "pageName": "Page 4",
    "term": "El Con Marriott",
  },
  Object {
    "definition": "British exclamation meaning first-rate or excellent",
    "pageName": "Page 4",
    "term": "top-hole",
  },
  Object {
    "definition": "A fictional junior tennis tournament, sponsored by Whataburger®, a real fast-food chain in the southwest U.S. See p. 14.",
    "pageName": "Page 4",
    "term": "WhataBurger Southwest Junior Invitational",
  },
  Object {
    "definition": "Another way of saying, \\"It ain't over till the fat lady sings.\\" This expression refers to opera, particularly those by Richard Wagner.",
    "pageName": "Page 5",
    "term": "\\"...the fat women in the Viking hat having sung...\\"",
  },
  Object {
    "definition": "Out of eight people in the room (including Hal, three deans, the Director of Composition, the varsity tennis coach, deLint, and C.T.), five are looking at Hal. Hal not being able to look at himself, two people are not looking at Hal, presumably deLint and C.T.",
    "pageName": "Page 5",
    "term": "62.5%",
  },
  Object {
    "definition": "Presumably Edmonton, Alberta.",
    "pageName": "Page 5",
    "term": "Edmonton",
  },
  Object {
    "definition": "Spots of color.",
    "pageName": "Page 5",
    "term": "mottle",
  },
  Object {
    "definition": "A circumflex is a diacritical mark, as seen in the French verb être (to be). Presumably, the dean's eyebrows have taken on this shape.",
    "pageName": "Page 5",
    "term": "circumflex",
  },
  Object {
    "definition": "The Pacific 10 athletic conference, the other members of which are: Arizona State Univ., Univ. of California at Berkeley, Univ. of Oregon, Oregon State Univ., Stanford Univ., UCLA, USC, Univ. of Washington, and Washington State Univ.",
    "pageName": "Page 5",
    "term": "Pac 10",
  },
  Object {
    "definition": "\\"Kekuléan\\" is not a type of knot. To Hal, the knot he is focusing on resembles the self-consuming, annular shape of the snake that inspired August Kekulé's discovery of benzene's molecular structure. August Kekule (1829-1896), a renowned German organic chemist, was the principal founder of the theory of chemical structure.  His most famous work, the discovery of benzene molecule's structure, is said to be inspired by a dream.  \\"Kekulé's Dream\\" was that of a self-devouring snake, the shape of which he used to describe the benzene ring. Hal's intense focus on this annular, or ring-like, part of the tie is the first reference to annular shapes.
        ",
    "pageName": "Page 5",
    "term": "\\"I stare carefully into the Kekuléan knot of the middle Dean's necktie.\\"",
  },
  Object {
    "definition": "Judging from the way the words sound when spoken.",
    "pageName": "Page 6",
    "term": "phonetic perspective",
  },
  Object {
    "definition": "This word, not found in dictionaries, would seem to mean \\"of or pertaining to an aviary,\\" an aviary being where birds are kept.",
    "pageName": "Page 6",
    "term": "aviarian",
  },
  Object {
    "definition": "\\"Marked by conciseness, precision, or refinement of expression: lapidary prose\\" (thefreedictionary.com). The OED defines it as “Characteristic of or suitable for monumental inscriptions”. The original meaning refers to the cutting and polishing of precious stones. Wallace favored this word to describe well-wrought prose, and used it often himself in interviews and readings.",
    "pageName": "Page 7",
    "term": "lapidary",
  },
  Object {
    "definition": "Overrefined.",
    "pageName": "Page 7",
    "term": "effete",
  },
  Object {
    "definition": "This term describes a school of thought that there are rules of grammar that should be obeyed and taught. Wallace wrote at length about the thorny questions surrounding this subject in the famous essay, \\"Tense Present:  Democracy, English, and the Wars over Usage,\\" which can be found at http://www.harpers.org/archive/2001/04/0070913",
    "pageName": "Page 7",
    "term": "Prescriptive Grammar",
  },
  Object {
    "definition": "Named for Jean Baptiste Joseph Fourier (1768-1830), a French mathematician, a Fourier transformation is \\"a certain linear operator that maps functions to other functions\\" (Wikipedia). Post-Fourier would refer to those transformations that came after Fourier.",
    "pageName": "Page 7",
    "term": "Post-Fourier Transformations",
  },
  Object {
    "definition": "Approximating reality using holograms.",
    "pageName": "Page 7",
    "term": "Holographically Mimetic",
  },
  Object {
    "definition": "Inactivity resulting from a static balance between opposing forces [1].",
    "pageName": "Page 7",
    "term": "Stasis",
  },
  Object {
    "definition": "Named for Richard Merett Montague (1930-1971), an American logician, this is an approach to semantics that suggests that the semantics of natural languages is essentially the same as those of formal languages, such as logic or computer programming.",
    "pageName": "Page 7",
    "term": "Montague Grammar",
  },
  Object {
    "definition": "Modality, in linguistics, refers to sign theory. Physical modality would, therefore, be either how a physical thing is represented by a sign or how any idea is represented by something physical.",
    "pageName": "Page 7",
    "term": "Physical Modality",
  },
  Object {
    "definition": "Third-level, after primary and secondary.",
    "pageName": "Page 7",
    "term": "Tertiary",
  },
  Object {
    "definition": "The era of the reign of Byzantine Emperor Justinian I (reigned 527-565).",
    "pageName": "Page 7",
    "term": "Justinian",
  },
  Object {
    "definition": "Italian for \\"below.\\"  Particularly when used in the phrase, \\"sotto voce,\\" it means speaking in a low voice, under one's breath.",
    "pageName": "Page 7",
    "term": "sotto",
  },
  Object {
    "definition": "Oxbridge refers to the two oldest colleges in the U.K., Oxford and Cambridge. The Quadrivium are the four academic subjects of arithmetic, geometry, music and astronomy. The Trivium are three disciplines, i.e., grammar, logic, and rhetoric.",
    "pageName": "Page 8",
    "term": "Oxbridge Quadrivium-Trivium",
  },
  Object {
    "definition": "Growth of tissue, especially muscle.  Although there are many causes, the most common is exercise.  (see, contra, atrophied.)",
    "pageName": "Page 8",
    "term": "hypertrophied",
  },
  Object {
    "definition": "A neologism, meaning infused with insignia (a distinguishing mark or sign, many graphic logos are insignia).",
    "pageName": "Page 9",
    "term": "insigniated",
  },
  Object {
    "definition": "North American Associaton of University Professors, the presumed follower to the American Assocation of University Professors.",
    "pageName": "Page 9",
    "term": "N.A.A.U.P.",
  },
  Object {
    "definition": "French: from me.",
    "pageName": "Page 9",
    "term": "de moi",
  },
  Object {
    "definition": "\\"Whosoever\\" would be the proper subjective form of this word. Hal is saying that the Deans, even with their limited grammatical abilities, would find the recent essays appalling.",
    "pageName": "Page 9",
    "term": "\\"...who use whomsoever as a subject...\\"",
  },
  Object {
    "definition": "one hip lower than the other.",
    "pageName": "Page 9",
    "term": "hip-shot",
  },
  Object {
    "definition": "The smallest networks of blood vessels, where arteries turn into veins.",
    "pageName": "Page 9",
    "term": "capillary webs",
  },
  Object {
    "definition": "As if eliminating solid bodily waste.",
    "pageName": "Page 9",
    "term": "defacatory",
  },
  Object {
    "definition": "A mafia boss.",
    "pageName": "Page 9",
    "term": "Don",
  },
  Object {
    "definition": "An acronym for the Racketeer Influenced and Corrupt Organizations Act, a statute used primarily to charge organized crime figures in criminal conspiracies.",
    "pageName": "Page 9",
    "term": "RICO",
  },
  Object {
    "definition": "whirling",
    "pageName": "Page 10",
    "term": "vortexing",
  },
  Object {
    "definition": "relating to the practice of favoring relatives or friends",
    "pageName": "Page 10",
    "term": "nepotistic",
  },
  Object {
    "definition": "Named for Sir David Brewster (1781-1868), Scottish scientist; the angle at which non-polarized light striking a surface will reflect polarized light. Presumably a desk lamp is positioned at such an angle. For more.",
    "pageName": "Page 10",
    "term": "Brewster's-Angle light",
  },
  Object {
    "definition": "a neologism by Hal's criteria, also present in urbandictionary only as a noun or adjective. Perhaps the intended meaning is \\"dicked with.\\"",
    "pageName": "Page 10",
    "term": "dickied",
  },
  Object {
    "definition": "A suburb of Boston, about 17 miles west of the city",
    "pageName": "Page 10",
    "term": "Weston",
  },
  Object {
    "definition": "The passage in which Hal eats the mold is reproduced not quite verbatim from DFW's purportedly autobiographical 1991 Harper's Magazine essay \\"Tennis, Trigonometry, Tornadoes: A Midwestern Boyhood\\" p. 70.",
    "pageName": "Page 10",
    "term": "...Popsicle sticks and twine.",
  },
  Object {
    "definition": "Hal's older brother and the middle name of Hal & Orin's father, James O. Incandenza, literally \\"a tree\\" or \\"pale.\\"",
    "pageName": "Page 10",
    "term": "Orin",
  },
  Object {
    "definition": "a brand name of rotary tiller",
    "pageName": "Page 10",
    "term": "Rototiller",
  },
  Object {
    "definition": "Children's clothing bearing Winnie-the-Pooh cartoon images or graphics, presumably pajamas in this instance",
    "pageName": "Page 10",
    "term": "Pooh-wear",
  },
  Object {
    "definition": "hairy",
    "pageName": "Page 10",
    "term": "hirsute",
  },
  Object {
    "definition": "Literally \\"old-eyed,\\" this is the inability to focus one's eyes as one grows older",
    "pageName": "Page 11",
    "term": "presbyopic",
  },
  Object {
    "definition": "A Wallace neologism (and portmanteaux) to describe the effects (shaking hands) of prolonged operation of a Rototiller",
    "pageName": "Page 11",
    "term": "Rototrembling",
  },
  Object {
    "definition": "adj., biting, stinging, sarcastic",
    "pageName": "Page 11",
    "term": "caustic",
  },
  Object {
    "definition": "adj., straight or true; in line with",
    "pageName": "Page 11",
    "term": "plumb",
  },
  Object {
    "definition": "war-like",
    "pageName": "Page 11",
    "term": "martial",
  },
  Object {
    "definition": "an idea represented by a shape, e.g., a stop sign, known by its eight-sided configuration",
    "pageName": "Page 11",
    "term": "ideogram",
  },
  Object {
    "definition": "ROM is an acronym for \\"Read Only Memory\\", a class of computer data storage. In the real world, best known in the name of the non-music version of Compact Discs (CD-ROM). CD-ROMs were becoming a popular way to distribute software (and pre-Internet computerized encyclopedias and atlases) when Infinite Jest was written, and even then it was predicted that DVD-ROMs or some other video/data disk would eventually supplant them. In more technical contexts, ROM refers to a specific variety of computer chips, but since Hal is talking about \\"drives\\", it seems likely that he means something more like a CD.",
    "pageName": "Page 12",
    "term": "ROM-drives",
  },
  Object {
    "definition": "Søren Aabye Kierkegaard was a 19th century Danish philosopher and one of the progenitors of existential philosophy",
    "pageName": "Page 12",
    "term": "Kierkegaard",
  },
  Object {
    "definition": "Albert Camus was a 20th century Algerian-born French author of existentialist texts.",
    "pageName": "Page 12",
    "term": "Camus",
  },
  Object {
    "definition": "Dennis Gabor, born Gábor Dénes, was a 20th century Hungarian physicist who invented holography, for which he received the Nobel Prize.",
    "pageName": "Page 12",
    "term": "Dennis Gabor",
  },
  Object {
    "definition": "Thomas Hobbes (1588-1679) was a British philosopher and author of Leviathan. In it, he suggests that the only escape from living in a state of nature that is \\"solitary, nasty, brutish, and short\\" is to build societies. Jean-Jacques Rousseau (1712-1778) was the Swiss-French philosopher who wrote The Social Contract, in which he advances the same argument but idealizes the state of nature.",
    "pageName": "Page 12",
    "term": "\\"...Hobbes is just Rousseau in a dark mirror...\\"",
  },
  Object {
    "definition": "Georg Wilhelm Friedrich Hegel (1770-1831) was a highly influential German philosopher.",
    "pageName": "Page 12",
    "term": "Hegel",
  },
  Object {
    "definition": "Latin for \\"creation,\\" the line over the a indicates the vowel is long and pronounced as in \\"father\\" rather than in \\"hate.\\"",
    "pageName": "Page 12",
    "term": "creatus",
  },
  Object {
    "definition": "v. tr., \\"To restrain or immobilize (a person) by binding the arms\\" (thefreedictionary.com)",
    "pageName": "Page 12",
    "term": "pinion",
  },
  Object {
    "definition": "An in-laid wood pattern, often a block-pattern, typically in flooring. Also, in France, the branch of the law that deals with the persecution of crime.",
    "pageName": "Page 12",
    "term": "parquet",
  },
  Object {
    "definition": "Note: The same words used by the Moms when Hal ate the mold.",
    "pageName": "Page 12",
    "term": "\\"God! Help!\\"",
  },
  Object {
    "definition": "A brand of shoes, generally pricey.",
    "pageName": "Page 12",
    "term": "Nunn Bush",
  },
  Object {
    "definition": "a wrestling hold with the offensive competitor's arm wrapped under the opponent's arm and over the opponent's neck from behind, allowing an opponent to be immobilized or levered from behind",
    "pageName": "Page 13",
    "term": "half nelson",
  },
  Object {
    "definition": "The Heimlich maneuver, named for contemporary American physician Henry Jay Heimlich, dislodges food from a choking person's trachea by applying sharp pressure to the abdomen.",
    "pageName": "Page 13",
    "term": "Heimlich",
  },
  Object {
    "definition": "To move about in whirling manner.",
    "pageName": "Page 13",
    "term": "roil",
  },
  Object {
    "definition": "This is the plural of pase, a Spanish word used in bullfighting to denote the movement of the matador's cape in drawing in the bull.",
    "pageName": "Page 13",
    "term": "pases",
  },
  Object {
    "definition": "lying on one's back",
    "pageName": "Page 13",
    "term": "supine",
  },
  Object {
    "definition": "a word used to denote a type of military gunfire.  A formation or position is \\"in enfilade\\" if weapons fire can be directed along its length. For instance, a column of marching troops is enfiladed if fired on from the front rather than the side.",
    "pageName": "Page 13",
    "term": "enfilade",
  },
  Object {
    "definition": "Synecdoche for the the fictional \\"WhataBurger Southwest Junior Invitational,\\" an annual juniors' tennis tournament held in the novel in Tucson, AZ.  (A Corpus Christi invention, Whataburger® is a well-established local burger chain in Phoenix (with 28 franchisees in AZ in 2009) but whose real fan base hails from Texas.)
        ",
    "pageName": "Page 14",
    "term": "Whataburger",
  },
  Object {
    "definition": "Sticky, thick and liquid.",
    "pageName": "Page 14",
    "term": "viscous",
  },
  Object {
    "definition": "shoes popular in Latin America with rope for soles",
    "pageName": "Page 15",
    "term": "espadrilles",
  },
  Object {
    "definition": "a savant is a person of learning, particularly specialized knowledge of a particular field.",
    "pageName": "Page 15",
    "term": "savant",
  },
  Object {
    "definition": "To shove.",
    "pageName": "Page 15",
    "term": "shunt",
  },
  Object {
    "definition": "Alludes to kangaroo-court, a sham legal proceeding.",
    "pageName": "Page 15",
    "term": "kangaroo-interview",
  },
  Object {
    "definition": "lion-like",
    "pageName": "Page 15",
    "term": "leonine",
  },
  Object {
    "definition": "plural of cirrus, a type of cloud",
    "pageName": "Page 15",
    "term": "cirri",
  },
  Object {
    "definition": "Direct paths to desired locations",
    "pageName": "Page 15",
    "term": "vectors",
  },
  Object {
    "definition": "a strict disciplinarian",
    "pageName": "Page 15",
    "term": "martinet",
  },
  Object {
    "definition": "Named for Ernst Mach (1838-1916), a Bohemian-Austrian physicist, the mach unit is a unit for the speed of sound. \\"Ultra-mach\\" would apply to a plane flying at several times the speed of sound.",
    "pageName": "Page 16",
    "term": "ultra-mach",
  },
  Object {
    "definition": "This word refers to \\"aged and weathered boards, esp. those salvaged from dismantled barns\\" (Random House Unabridged Dictionary).",
    "pageName": "Page 16",
    "term": "barnwood",
  },
  Object {
    "definition": "Employing the nautical term for \\"right\\" (starboard), the woman referred to tends to move right as she tries to move forward.",
    "pageName": "Page 16",
    "term": "starboard list",
  },
  Object {
    "definition": "excessive or abnormally large growth in humans, also giantism",
    "pageName": "Page 16",
    "term": "gigantism",
  },
  Object {
    "definition": "having the qualities of a parody",
    "pageName": "Page 16",
    "term": "parodic",
  },
  Object {
    "definition": "one subject to infantophilia (see pedophilia), but may simply reference the earlier Inner Infant group",
    "pageName": "Page 16",
    "term": "infantophile",
  },
  Object {
    "definition": "a neologism meaning \\"of or like an incision\\"",
    "pageName": "Page 16",
    "term": "incisionish",
  },
  Object {
    "definition": "Wallace neologism describing a smallness of fingers or hands [hypophalangeal would be the proper spelling, by analogy to interphalangeal, an extant word, used to describe two of the three joints in which the phalanges, or segments, of he fingers and toes participate, excluding the thumb and great toe, which have only one interphalangeal joint.]",
    "pageName": "Page 16",
    "term": "hypophalangial",
  },
  Object {
    "definition": "An Incandenza family nickname for Hal's father, James O. Incandenza; the first reference to James O. Incandenza in the novel",
    "pageName": "Page 16",
    "term": "Himself's",
  },
  Object {
    "definition": "Referring here to the portable phone's antenna.",
    "pageName": "Page 16",
    "term": "antenna",
  },
  Object {
    "definition": "See Hamlet, Act Five, Scene One.",
    "pageName": "Page 17",
    "term": "\\"...Donald Gately and I dig up my father's head...\\"",
  },
  Object {
    "definition": "At the time this novel was published, Venus Williams would have been sixteen years old.",
    "pageName": "Page 17",
    "term": "Venus Williams",
  },
  Object {
    "definition": "Dymphna was a 7th century Irish saint. Her feast day is May 15. She is the patron saint of mental illness professionals, epileptics, and the mentally ill, among others.",
    "pageName": "Page 17",
    "term": "Dymphna",
  },
  Object {
    "definition": "Petropolis is Greek for \\"city of stone\\" or \\"city of Peter.\\" Petrópolis is a city in Brazil near Rio de Janeiro. A treaty was signed there on November 11, 1903, ending hostilities between Bolivia and Brazil.
\\"Kahn\\" is a variant on the Jewish name for a priest, i.e., kohen.
        ",
    "pageName": "Page 17",
    "term": "Petropolis Kahn",
  },
  Object {
    "definition": "the cause of a disease",
    "pageName": "Page 17",
    "term": "etiology",
  },
  Object {
    "definition": "Adjective meaning either relating to the surgical restoration of a lost or absent part or
          relating to or characterized by cells that have become less differentiated/primitive.
        ",
    "pageName": "Page 17",
    "term": "anaplastic",
  },
  Object {
    "definition": "a technique of teaching by asking students questions, attributed to Socrates' pedagogy in ancient Greece",
    "pageName": "Page 17",
    "term": "Socratic method",
  },
  Object {
    "definition": "This is a reference to the Oxford English Dictionary, sixth edition.",
    "pageName": "Page 17",
    "term": "O.E.D. VI's count",
  },
  Object {
    "definition": "still in use, as distinct from those dictionary words considered archaic and not part of the modern language",
    "pageName": "Page 17",
    "term": "nonarchaic",
  },
  Object {
    "definition": "deriving from Latin",
    "pageName": "Page 17",
    "term": "Latinate",
  },
  Object {
    "definition": "deriving from Old English",
    "pageName": "Page 17",
    "term": "Saxonic",
  },
  Object {
    "definition": "Wallace neologism for \\"bitten to the quick\\", as in nails gnawed down to where they emerged from the fingertips",
    "pageName": "Page 17",
    "term": "quick-bit",
  },
  Object {
    "definition": "the sound of a Spanish-speaker's pronunciation of \\"you\\"",
    "pageName": "Page 17",
    "term": "jou",
  },
  Object {
    "definition": "An upright beam",
    "pageName": "Page 17",
    "term": "girder",
  },
  Object {
    "definition": "a little over seven ounces",
    "pageName": "Page 18",
    "term": "200 grams",
  },
  Object {
    "definition": "The implication here is that in the time of the book, there are videophones.",
    "pageName": "Page 18",
    "term": "\\"...using just audio...\\"",
  },
  Object {
    "definition": "A part of Boston proper, west of downtown and across the Charles river from Cambridge. The fictional Enfield most likely occupies part of what is in reality Allston.",
    "pageName": "Page 18",
    "term": "Allston",
  },
  Object {
    "definition": "generally high-quality marijuana, containing a high volume of resins where THC in marijuana plants is produced",
    "pageName": "Page 18",
    "term": "high-resin dope",
  },
  Object {
    "definition": "vernacular, arguably offensive, term for a cleft lip",
    "pageName": "Page 18",
    "term": "harelip",
  },
  Object {
    "definition": "\\"Teleputer\\", as used elsewhere in the text. Assumed to be a hybridized communications/entertainment device.",
    "pageName": "Page 19",
    "term": "TP",
  },
  Object {
    "definition": "To physically possess marijuana.",
    "pageName": "Page 19",
    "term": "\\"own marijuana\\"",
  },
  Object {
    "definition": "used as a verb, communicating with the office via modem, an early but ubiquitous tool for transmitting data between servers and client/servers. The word is a portmanteau of modulate / demodulate.",
    "pageName": "Page 19",
    "term": "modem",
  },
  Object {
    "definition": "electronic note, likely not a literal reference to an actual electronic communication, conceived in the pre-Internet era … not sure what this means, DFW may have intended some form of communication different from email, which had been in progressively increasingly widespread use since he was in elementary school.",
    "pageName": "Page 19",
    "term": "e-note",
  },
  Object {
    "definition": "a member of the Royal Canadian Mounted Police, their equivalent of the FBI",
    "pageName": "Page 20",
    "term": "Mountie",
  },
  Object {
    "definition": "a neighborhood of Cambridge bordering on Somerville, about a mile from Tufts University, which is on the Somerville/Medford border",
    "pageName": "Page 20",
    "term": "Porter Square, Cambridge",
  },
  Object {
    "definition": "Experiencing convulsions, violent involuntary physical shaking",
    "pageName": "Page 21",
    "term": "convulsive",
  },
  Object {
    "definition": "This would presumably be a festival celebrating the plays of German playwright Benjamin Franklin Wedekind (1864-1914), a proto-expressionist.",
    "pageName": "Page 21",
    "term": "Wedekind festival",
  },
  Object {
    "definition": "Aggressively greedy.",
    "pageName": "Page 21",
    "term": "rapacious",
  },
  Object {
    "definition": "inflammation of the pleurae, the membranes surrounding the lungs",
    "pageName": "Page 22",
    "term": "pleurisy",
  },
  Object {
    "definition": "A television-based home entertainment system, ubiquitous in the time the novel is set, which plays copy-protected \\"cartridges\\" custom-ordered by viewers, invented by Noreen Lace-Forché",
    "pageName": "Page 22",
    "term": "Interlace viewer",
  },
  Object {
    "definition": "with intense attention to",
    "pageName": "Page 22",
    "term": "raptly",
  },
  Object {
    "definition": "See Interlace viewer, above",
    "pageName": "Page 22",
    "term": "cartridge",
  },
  Object {
    "definition": "an episode of debauchery, engaging in excessive, pleasure-seeking, often sexual- or drug/alcohol-related",
    "pageName": "Page 22",
    "term": "debauch",
  },
  Object {
    "definition": "about 4.2 ounces",
    "pageName": "Page 22",
    "term": "120 grams",
  },
  Object {
    "definition": "of low character and lacking integrity",
    "pageName": "Page 22",
    "term": "debased",
  },
  Object {
    "definition": "Ernest Anthony Puente, Jr., an internationally known Puerto Rican jazz musician.",
    "pageName": "Page 22",
    "term": "Tito Puente",
  },
  Object {
    "definition": "Marlborough Street runs through the Back Bay area of Boston.",
    "pageName": "Page 22",
    "term": "Marlborough Street",
  },
  Object {
    "definition": "taking something as one's own, without permission",
    "pageName": "Page 23",
    "term": "appropriation",
  },
  Object {
    "definition": "As the endnote on p. 983 tells us, this the chemical name for crystal meth. Calling to mind that Infinite Jest was published in 1996, don't think crystal meth is a new phenomenon.",
    "pageName": "Page 23",
    "term": "methamphetamine hydrochloride",
  },
  Object {
    "definition": "describing a work that imitates the style of another work, artist, movement or period",
    "pageName": "Page 24",
    "term": "pastiche",
  },
  Object {
    "definition": "typo or intentional misspelling of majuscule, an initial capital letter, often large-type to introduce a section of written material; Wallace's use of the noun in a verb form is likely a neologism (especially if the alternate spelling is retained)",
    "pageName": "Page 24",
    "term": "magisculed",
  },
  Object {
    "definition": "about 1.75 ounces",
    "pageName": "Page 24",
    "term": "50 grams",
  },
  Object {
    "definition": "grown in water without soil",
    "pageName": "Page 24",
    "term": "hydroponic",
  },
  Object {
    "definition": "a drinking mug",
    "pageName": "Page 25",
    "term": "stein",
  },
  Object {
    "definition": "acronym for \\"Empire Waste Disposal\\"; a garbage truck",
    "pageName": "Page 25",
    "term": "E.W.D. land barge",
  },
  Object {
    "definition": "biased from a male point of view",
    "pageName": "Page 25",
    "term": "phallocentric",
  },
  Object {
    "definition": "nearly 20 inches",
    "pageName": "Page 25",
    "term": "half a meter",
  },
  Object {
    "definition": "Short for \\"carburetor,\\" just as the carburetor in an internal combustion engine mixes air with gas to allow combustion, the carburetor on a water pipe allows one to draw air in with marijuana smoke.",
    "pageName": "Page 25",
    "term": "carb",
  },
  Object {
    "definition": "slanted, like an acute or obtuse angle, not a right-angle",
    "pageName": "Page 25",
    "term": "oblique",
  },
  Object {
    "definition": "Combination television and computer, generic term for an Interlace player, see also TP",
    "pageName": "Page 25",
    "term": "teleputer",
  },
  Object {
    "definition": "correspond with; match up",
    "pageName": "Page 26",
    "term": "jibe",
  },
  Object {
    "definition": "indirect or dishonest",
    "pageName": "Page 26",
    "term": "oblique",
  },
  Object {
    "definition": "dried out",
    "pageName": "Page 27",
    "term": "desiccated",
  },
  Object {
    "definition": "As if struck by a convulsion; moving suddenly and without coordination.",
    "pageName": "Page 27",
    "term": "convulsively",
  },
]
`;

exports[`WallaceWikiParser should be able to parse table of contents 1`] = `
Array [
  "/david-foster-wallace/index.php?title=Pages_3-27",
  "/david-foster-wallace/index.php?title=Pages_27-63",
  "/david-foster-wallace/index.php?title=Pages_63-87",
  "/david-foster-wallace/index.php?title=Pages_87-127",
  "/david-foster-wallace/index.php?title=Pages_127-156",
  "/david-foster-wallace/index.php?title=Pages_157-181",
  "/david-foster-wallace/index.php?title=Pages_181-198",
  "/david-foster-wallace/index.php?title=Pages_198-219",
  "/david-foster-wallace/index.php?title=Pages_219-258",
  "/david-foster-wallace/index.php?title=Pages_258-283",
  "/david-foster-wallace/index.php?title=Pages_283-306",
  "/david-foster-wallace/index.php?title=Pages_306-321",
  "/david-foster-wallace/index.php?title=Pages_321-342",
  "/david-foster-wallace/index.php?title=Pages_343-379",
  "/david-foster-wallace/index.php?title=Pages_380-398",
  "/david-foster-wallace/index.php?title=Pages_398-418",
  "/david-foster-wallace/index.php?title=Pages_418-442",
  "/david-foster-wallace/index.php?title=Pages_442-469",
  "/david-foster-wallace/index.php?title=Pages_470-489",
  "/david-foster-wallace/index.php?title=Pages_489-508",
  "/david-foster-wallace/index.php?title=Pages_508-530",
  "/david-foster-wallace/index.php?title=Pages_531-562",
  "/david-foster-wallace/index.php?title=Pages_563-588",
  "/david-foster-wallace/index.php?title=Pages_589-619",
  "/david-foster-wallace/index.php?title=Pages_620-651",
  "/david-foster-wallace/index.php?title=Pages_651-662",
  "/david-foster-wallace/index.php?title=Pages_663-686",
  "/david-foster-wallace/index.php?title=Pages_686-698",
  "/david-foster-wallace/index.php?title=Pages_698-716",
  "/david-foster-wallace/index.php?title=Pages_716-735",
  "/david-foster-wallace/index.php?title=Pages_736-755",
  "/david-foster-wallace/index.php?title=Pages_755-785",
  "/david-foster-wallace/index.php?title=Pages_785-808",
  "/david-foster-wallace/index.php?title=Pages_809-827",
  "/david-foster-wallace/index.php?title=Pages_827-845",
  "/david-foster-wallace/index.php?title=Pages_845-876",
  "/david-foster-wallace/index.php?title=Pages_876-883",
  "/david-foster-wallace/index.php?title=Pages_883-902",
  "/david-foster-wallace/index.php?title=Pages_902-916",
  "/david-foster-wallace/index.php?title=Pages_916-934",
  "/david-foster-wallace/index.php?title=Pages_934-964",
  "/david-foster-wallace/index.php?title=Pages_964-981",
  "/david-foster-wallace/index.php?title=Notes_and_Errata_-_Pages_983-1079",
]
`;
